crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -2626,6 +2635,126 @@ impl Backend {
         Ok(GethTrace::Default(Default::default()))
     }
 
+    #[cfg(feature = "js-tracer")]
+    pub async fn trace_tx_with_js_tracer(
+        &self,
+        hash: B256,
+        code: String,
+        opts: GethDebugTracingOptions,
+    ) -> Result<GethTrace, BlockchainError> {
+        let GethDebugTracingOptions { tracer_config, .. } = opts;
+
+        let (info, block) = {
+            let storage = self.blockchain.storage.read();
+            let MinedTransaction { info, block_hash, .. } =
+                storage.transactions.get(&hash).unwrap().clone();
+            let block = storage.blocks.get(&block_hash).cloned().unwrap();
```

Comment:
here we'd need to handle the case where this doesnt exist and return an error instead of unwrapping